### PR TITLE
Ordinal transaction functions

### PR DESCRIPTION
**Pull request by abdulhaseeb4239**
# 🔘 PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Enhancement
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

# 📜 Background
We needed to move all the ordinals transactions logic to xverse-core. Including fetching and address UTXOs and getting the correct UTXO to send. 

Issue Link: https://linear.app/xverseapp/issue/ENG-2528/move-ordinal-transactions-logic-to-xverse-core
Context Link (if applicable):

# 🔄 Changes
I have created new functions to wrap up the existing ones. I passed the ordinal inscription and `isRecover` boolean. and function fetches the addressUtxos from ordinalAddress and from btcAddress if it's a recover transaction. 

- [ ] Yes, Incompatible API changes
- [x] No, Adds functionality (backwards compatible)
- [ ] No, Bug fixes (backwards compatible)

Changes:
- modified transactions/btc.ts file to add new functions for getting fees for ordinal send transactions and signing ordinal transactions. 

Impact:
- This will help keep the logic consistent across the app and extension. and future changes will be much easier. 
- Impact on downstream apps, link to issues or PRs for migration (TODO)
- Ordinals Transfer and Ordinal Recover

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: transactions/btc.ts [hunk 1]
```diff
@@ -469,6 +469,30 @@ export async function getBtcFeesForOrdinalSend(
...
+export async function getBtcFeesForOrdinalTransaction(
+  recipientAddress: string,
+  btcAddress: string,
+  ordinalsAddress: string,
+  network: NetworkType,
+  ordinal: Inscription,
+  isRecover?: boolean,
+  feeMode?: string,
+  feeRateInput?: string,
+) {
```
Comment (teebszet) [inline transactions/btc.ts:481]: can we use named params here?

File: transactions/btc.ts [hunk 2]
```diff
@@ -789,6 +813,40 @@ export async function signOrdinalSendTransaction(
...
+export async function signOrdinalTransaction(
+  recipientAddress: string,
+  btcAddress: string,
+  ordinalsAddress: string,
+  accountIndex: number,
+  seedPhrase: string,
+  network: NetworkType,
+  ordinal: Inscription,
+  fee?: BigNumber,
+  isRecover?: boolean,
```
Comment (teebszet) [inline transactions/btc.ts:825]: same here

File: transactions/btc.ts [hunk 3]
```diff
@@ -789,6 +813,40 @@ export async function signOrdinalSendTransaction(
...
+): Promise<SignedBtcTx> {
+  const btcClient = new BitcoinEsploraApiProvider({
+    network,
+  });
+  const address = isRecover ? btcAddress : ordinalsAddress;
+  const addressUtxos = await btcClient.getUnspentUtxos(address);
+  const ordUtxo = addressUtxos.find((utx) => `${utx.txid}:${utx.vout}` === ordinal.output);
```
Comment (teebszet) [inline transactions/btc.ts:832]:
```suggestion
  const ordUtxo = addressUtxos.find((utxo) => `${utxo.txid}:${utxo.vout}` === ordinal.output);
```

**Review by teebszet — APPROVED** [review]
lgtm

**Comment by teebszet** [comment]
@abdulhaseeb4239 can we merge this, since it's a non-breaking change? can do more testing on the web-extension and mobile PRs

**Comment by abdulhaseeb4239** [comment]
> @abdulhaseeb4239 can we merge this, since it's a non-breaking change? can do more testing on the web-extension and mobile PRs

yes we can merge

**Pull request merged**